src/optimization.jl

```diff
@@ -0,0 +1,17 @@
+function newton_raphson(fgh!, x::AbstractArray{T}; ρ = one(T)/2, c = 1e-4, tol = 1e-4, max_iter = 1000) where T
+    grad = zeros(T, length(x))
+    hes = zeros(T, length(x),length(x))
+    y = zero(T)
+    for i = 1:max_iter
+        y = fgh!(x, grad, hes, true)
+        search_dir = -(cholfact(Positive, hes)\grad)
+        (norm(search_dir) > tol) || return x, y, grad, hes
+        step_size = one(T)
+        while fgh!(x+search_dir*step_size, grad, hes, false) > y+c*step_size*(grad' * search_dir)
+            step_size *= ρ
+            step_size > 1e-10 || error("Linesearch failed! Problematic Hessian or Gradient?")
```

Comment:
We could get coverage to 100% if we could come up with a test that hits this as well as the `ConvergenceException` below. Is that doable?

Reply:
Sure, I'll add a small testset for the Newton algorithm tomorrow. How does one test for an error though?

Reply:
`@test_throws`

Reply:
Sounds great. Other than that, I think this is good to go. You've done such excellent work here!!